src/Base/Exception.h

```diff
@@ -33,6 +33,13 @@
 #include "FileInfo.h"
 #include "BaseClass.h"
 
+/* MACROS FOR THROWING EXCEPTIONS */
+
+#define THROW(exception) throw Base::exception(__FILE__,__LINE__,__PRETTY_FUNCTION__);
+#define THROWM(exception, message) throw Base::exception(message,__FILE__,__LINE__,__PRETTY_FUNCTION__);

```

Comment:
To enforce the condition that the strings should be C-string literals, you could save a lot of typing by leaving the commas out in the macros so all the overridden ctors aren't needed (except I guess __LINE__ needs to be stringized).  C and C++ will concatenate string literals - it's totally valid to do eg `std::cerr<<"Something bad happened in "__FILE__ << std::endl;`